### PR TITLE
remembering last directory, gcode file extensions, unpackGCodeInstruction repair loading Gcode the character ; at the end of the line

### PR DESCRIPTION
**Pull request by jerabina**
…nd setting (LastPrefDir)

### CODE REVIEW

**Comment by jerabina** [comment]
and I also added gcode and ngc to the list of suffixes

File: polargraphcontroller.pde [hunk 1]
```diff
@@ -1106,17 +1111,24 @@ void loadVectorWithFileChooser()
...
+      
+      LastImageDir = fc.getCurrentDirectory();
+      
       if (returned == JFileChooser.APPROVE_OPTION) 
       {
         File file = fc.getSelectedFile();
         if (file.exists())
         {
           RShape shape = loadShapeFromFile(file.getPath());
+          println(shape);
```
Comment (euphy) [inline polargraphcontroller.pde]: debugging code

File: polargraphcontroller.pde [hunk 2]
```diff
@@ -149,6 +149,8 @@ List<String> machineMessageLog = new ArrayList<String>();
 List<PreviewVector> previewCommandList = new ArrayList<PreviewVector>();
 long lastCommandQueueHash = 0L;
 
+File LastImageDir = null;
```
Comment (euphy) [inline polargraphcontroller.pde]: The case is wrong on these two variables - should be lower-case first letter.

**Comment by euphy** [comment]
Thanks for the request - appreciated, and looks good (especially that remembering file paths - that irritates me so much!) except for a couple of stylistic things. The names of the new variables is the main thing, but have a look at the printlns you've left in and think about whether they should be permanently part of the app.

Also sorry for this being so slow :)

**Comment by jerabina** [comment]
Well, it's in the bag, I'm sorry, but I do not know how to get out, I wanted to correct other debug forgotten how I repaired loading Gcode (some generators Gcode a semicolon at the end of the line).
All that pull request failed me, next time I'll be careful what you are at home here editing.

**Pull request merged**